src/main/resources/public/spid/js/spid-button.js

```diff
@@ -0,0 +1,25 @@
+function closeOpenItems(provider) {
+  // close the dropdown if the user clicks on another dropdown
+  let dropdowns = document.querySelectorAll(".spid-idp-button");
+  dropdowns.forEach(function(openDropdown) {
+    if (!openDropdown.id.includes(provider)){
+      openDropdown.classList.remove("spid-idp-button-open");
+    }
+  });
+}
+
+function expandDropdown(provider) {
+  closeOpenItems(provider);
+  let dropdownMenu = document.getElementById("spid-dropdown-" + provider);
+  dropdownMenu.classList.toggle("spid-idp-button-open");
+}
+
+window.onclick = function (event) {
```

Comment:
binding an event to the whole window to close a modal is not the best approach. You should really add an overlay layer to the viewport to handle clicking outside.